MonoGame.Framework/Platform/Native/Video.Native.cs

```diff
@@ -3,13 +3,180 @@
 // file 'LICENSE.txt', which is part of this source code package.
 
 using System;
+using System.IO;
+using MonoGame.Interop;
+using Microsoft.Xna.Framework.Graphics;
+
 
 namespace Microsoft.Xna.Framework.Media;
 
 public sealed partial class Video : IDisposable
 {
-    private void PlatformInitialize()
+    private unsafe MGM_Video* _video;
+    private Texture2D[] _frameCache;
+    private bool _muted = false;
+    private float _volume = 1.0f;
+    private bool _looped = false;
+
+    private unsafe void PlatformInitialize()
+    {
+        var mediaFilePath = Path.Combine(TitleContainer.Location, FileName);
+
+        // TODO: We should expose the back buffer count so that
+        // video playback and optimize the number of textures created
+        // for video frames.
+        int cachedFrameNum = 2; // Game.Instance.GraphicsDevice.BackBufferCount;
+
+        // TODO: May be worth letting the video return texture
+        // format, so it can avoid CPU conversions when possible.
+
+        _video = MGM.Video_Create(mediaFilePath, cachedFrameNum, out var width, out var height, out var fps, out var duration);
```

Comment:
Using the same design here as with Song.

The plan is that video does all the playback work internally managing some textures and an audio stream from a playback thread.